sdk/core/Azure.Core/src/ChainingClassifier.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Implements chaining of classifiers for the general case where the end-of-chain
+    /// <see cref="ResponseClassifier"/> is not a <see cref="CoreResponseClassifier"/>.
+    /// <see cref="CoreResponseClassifier"/> is preferred due to its enhanced performance
+    /// characteristics.
+    /// The classifier chain is a series of <see cref="ResponseClassificationHandler"/> classifiers
+    /// followed by the "end-of-chain" <see cref="ResponseClassifier"/>.  The handlers are
+    /// added to the chain via <see cref="RequestContext"/> or <see cref="RequestOptions"/>,
+    /// and applied starting with the most recently added handler, then applying status code
+    /// classification, and finally by applying the "end-of-chain" classifier.
+    /// </summary>
+    internal class ChainingClassifier : ResponseClassifier
+    {
+        private ResponseClassificationHandler[]? _handlers;
+        private ResponseClassifier _endOfChain;
+
+        public ChainingClassifier((int Status, bool IsError)[]? statusCodes,
+            ResponseClassificationHandler[]? handlers,
+            ResponseClassifier endOfChain)
+        {
+            if (handlers != null)
+            {
+                AddClassifiers(handlers);
+            }
+
+            if (statusCodes != null)
+            {
+                StatusCodeHandler handler = new StatusCodeHandler(statusCodes);
+                AddClassifiers(new StatusCodeHandler[] { handler });
```

Comment:
Does the AddClassifiers method take an array or Span? If Span, this line of code should allocate a span on the stack as opposed to allocating an array.

Reply:
It takes an array, but it makes a copy, so it could take a Span I believe - is that correct?

Reply:
Yeah, if it makes a copy, it should take a span, ReadOnlySpan to be precise.